mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -37,7 +37,7 @@ import org.apache.spark.sql.types.{DataType, DoubleType, StructType}
  * Note: Marked as private and DeveloperApi since this may be made public in the future.
  */
 private[ml] trait DecisionTreeParams extends PredictorParams
-  with HasCheckpointInterval with HasSeed {
+  with HasCheckpointInterval with HasSeed with HasWeightCol {
 
```

Comment:
@imatiach-msft  Params `weightCol` and `minWeightFractionPerNode` are introduced `DecisionTreeParams` and also exposed to RF and GBT.
But RF and GBT do not support sample weighting for now. Is there any plan to support it? or we should put these params into `DecisionTreeRegressorParams` and `DecisionTreeClassifierParams`?

Reply:
"Is there any plan to support it?"
yes, definitely, we should support it eventually, will look into adding it when I get a chance.  There's already a JIRA ticket for that as well.

Reply:
(tagging @zhengruifeng )

Reply:
Thanks @imatiach-msft ! I just read corresponding tickets. 
My concern is that we need to support weighting in RF & GBT in 3.0.0, otherwise two unused params will be added.

Reply:
Could someone link to the JIRA ticket for class weights in RF / GBT? 

I've been trying to track it down. 

I found this : https://issues.apache.org/jira/browse/SPARK-9478

However, it is marked resolved. I see that a PR for class weights in decision trees was merged, however, I do not see where the RF / GBT feature is tracked nor implemented.

It is appreciated.

Reply:
You can see it's resolved as a duplicate, of https://issues.apache.org/jira/browse/SPARK-19591 for 3.0.0